### PR TITLE
MH-13124, Unify linting for JavaScript and HTML

### PR DESCRIPTION
**Pull request by lkiesow**
This adds a global ESLint and html-linter configuration which modules
can use for linting and style checks. Modules can overwrite these rules
if necessary.

Modules which currently integrate this are:

- lti
- runtime-info-ui
- runtime-info-ui-ng
- engage-paella-player
- admin-ui (partly)

Though the admin-ui module has still a few rules disabled and does not
use the HTML linter right now.

### CODE REVIEW

File: docs/checkstyle/eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  'extends': 'eslint:recommended',
+  'parserOptions': {
+    'ecmaVersion': 2015
+  },
+  'plugins': [
+    'header'
+  ],
+  'rules': {
+    // enforce Opencast's license header at the begin of the JavaScript files
```
Comment (staubesv) [inline docs/checkstyle/eslintrc.js]: the beginning?

File: docs/checkstyle/eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    ],
+    // no line ending with spaces:
+    // var valid;|<--
+    // var invalid;  |<--
+    'no-trailing-spaces': [
+      'error'
+    ],
+    // enforce one type of quotes:
+    // var x = 'valid';
+    // var y = 'invalid';
```
Comment (staubesv) [inline docs/checkstyle/eslintrc.js]: Maybe "invalid"? (double-quotes)

File: modules/admin-ui/src/main/webapp/index.html [hunk 3]
```diff
@@ -1,401 +1,402 @@
...
-                            <span translate="HELP.REST_DOC"><!-- REST Doc --></span>
-                        </a>
-                        </li>
-                        <li>
-                            <a translate="HELP.HOTKEY_CHEAT_SHEET" ng-click="showHotkeyCheatSheet()"></a>
-                        </li>
-                    </ul>
-                </div>
+        <div ng-if="documentationUrl || restdocsUrl" title="{{ 'HELP.HELP' | translate}}" class="nav-dd"
+                                                                                          id="help-dd" old-admin-ng-dropdown="">
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/index.html:76]: That seems weird, though... why not align to ng-if?

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 4]
```diff
@@ -1,429 +1,429 @@
...
+          <h4 class="video-details"><span class="video-details-date">{{ video.date | localizeDate : 'date' : 'short'}}</span>
+            <span class="video-details-presenters">
+              {{(video.presenters && video.presenters.length > 0)  ?  " - " + video.presenters.join(', ') : "" }}
+            </span>
+            <span class="video-details-series">{{(video.series && video.series.title) ? " - " + video.series.title : "" }}</span></h4>
+        </div>
+        <div class="video-save-panel video-more-options video-editor-actions-toolbar" ng-if="tab === 'editor' && player.adapter">
+          <div class="buttons">
+            <select chosen pre-select-from="video.workflows" class="workflow"
+                                                             ng-disabled="!video.workflows || video.workflows.length === 0"
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:30]: Again, seems weird...

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 5]
```diff
@@ -1,429 +1,429 @@
...
+        Uncomment this to see how tool selection works.
         <div class="tab-menu">
-            <nav class="view-controller">
-                <a ng-class="{active: tab === 'editor'}"
-                    ng-click="openTab('editor')" translate>VIDEO_TOOL.TABS.EDIT</a>
-                <a ng-Class="{active: tab === 'captions'}"
-                    ng-click="openTab('captions')" translate>VIDEO_TOOL.TABS.CAPTIONS</a>
-            </nav>
+        <nav class="view-controller">
+        <a ng-class="{active: tab === 'editor'}"
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:51]: Indentation

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 6]
```diff
@@ -1,429 +1,429 @@
...
+                        <i class="fa fa-exclamation-circle error"></i>
+                      </div>
+                    </div>
+
+                    <div class="controls">
+                      <div class="description">
+                        <p translate="VIDEO_TOOL.THUMBNAIL.DESCRIPTION"><!-- Description --></p>
+                        <br/>
+                        <p ng-if="video.thumbnail.type === 'DEFAULT'" translate="VIDEO_TOOL.THUMBNAIL.CURRENT.DEFAULT">
+                        <!-- You are currently using the default thumbnail. -->
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:132]: Indentation (same on lines downwards)

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 7]
```diff
@@ -1,429 +1,429 @@
...
+                <div class="full-col">
+                  <div class="obj comments">
+                    <header class="no-expand" translate="EVENTS.EVENTS.DETAILS.COMMENTS.CAPTION">
+                      <!-- Comments -->
+                    </header>
+                    <div class="obj-container">
+                      <div class="comment-container">
+                        <div class="comment-view">
+                          <div class="comment" ng-class="{ active : $parent.replyToId === comment.id }"
+                                               ng-repeat="comment in comments track by $index">
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:208]: Seems weird....

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 8]
```diff
@@ -1,429 +1,429 @@
...
+                            <textarea ng-model="myComment.text" placeholder="{{ 'EVENTS.EVENTS.DETAILS.COMMENTS.PLACEHOLDER' | translate }}"></textarea>
+                            <div class="reasons">
+                              <select chosen pre-select-from="components.eventCommentReasons"
+                                             data-width="'200px'"
+                                             data-disable-search-threshold="0"
+                                             ng-model="myComment.reason"
+                                             ng-options="value | translate for (id, value) in components.eventCommentReasons"
+                                             data-placeholder="{{ 'EVENTS.EVENTS.DETAILS.COMMENTS.SELECTPLACEHOLDER' | translate }}"
+                                             >
+                                             <option value=""></option>
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:253]: indentation

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 9]
```diff
@@ -1,429 +1,429 @@
...
-                                </div>
-
-                                <span ng-if="!video.source_tracks[$index].video.present"
-                                      class="video-line-height"
-                                      translate="VIDEO_TOOL.TRACKS.VIDEO.NOT_AVAILABLE">
-                                  <!-- Stream has no video -->
+            <div ng-if="area === 'tracks'" class="editor-tracks">
+              <div class="area-body">
+                <div data-admin-ng-notification="" data-type="error"
+                                                   show="!anyTrackSelected('video')" message="VIDEO_TOOL.TRACKS.NOTIFICATION.NO_VIDEO_TRACK_SELECTED">
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:315]: seems weird (why not align show to data-admin-ng-notification?)

File: modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html [hunk 10]
```diff
@@ -1,429 +1,429 @@
...
-                        <select chosen pre-select-from="video.workflows" class="workflow"
-                            ng-disabled="!video.workflows || video.workflows.length === 0"
-                            data-width="'175px'"
-                            data-disable-search-threshold="0"
-                            ng-model="video.workflow"
-                            ng-options="workflow.id as workflow.name for workflow in video.workflows | orderBy: 'displayOrder':true" />
+            <div class="video-save-panel" ng-if="tab === 'editor' && player.adapter">
+              <div class="buttons">
+                <select chosen pre-select-from="video.workflows" class="workflow"
+                                                                 ng-disabled="!video.workflows || video.workflows.length === 0"
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/modules/events/subresources/partials/tools.html:408]: Seems weird...

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html [hunk 11]
```diff
@@ -1,222 +1,223 @@
...
+
+  <div class="modal-content" ng-class="{active: currentForm === 'deleteForm'}" ng-if="currentForm === 'deleteForm'">
+
+    <div class="modal-body">
+
+      <div class="full-col">
+        <div class="list-obj" ng-form="deleteForm">
+          <div class="modal-alert danger obj" ng-if="events.unpublished.has">
+            <p translate="BULK_ACTIONS.DELETE_EVENTS_WARNING_LINE1">
+            <!-- You have chosen to delete an event. Once deleted all metadata,
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html:23]: Indentation of comments (same downwards)

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html [hunk 12]
```diff
@@ -1,222 +1,223 @@
...
+                    <td>{{ row.presenters }}</td>
+                  </tr>
+                </tbody>
+              </table>
+            </div> <!-- obj -->
+          </div><!-- full-col -->
+
+          <div class="full-col obj" ng-if="!events.unpublished.has">
+            <p translate="BULK_ACTIONS.DELETE.EVENTS.UNPUBLISHED.NONE">
+            <!-- None of the selected events can be deleted without leaving orphan files. -->
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html:63]: Indentation

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html [hunk 13]
```diff
@@ -1,222 +1,223 @@
...
+
+  <div class="modal-content" ng-class="{active: currentForm === 'retractForm'}">
+
+    <div class="modal-body">
+      <div class="full-col">
+        <div class="list-obj" ng-form="retractForm">
+
+          <div class="modal-alert danger obj" ng-if="events.published.has">
+            <p translate="BULK_ACTIONS.DELETE.EVENTS.PUBLISHED.WARNING">
+            <!-- Some events can not be deleted as they have already been published and deleting
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html:92]: Indentation

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html [hunk 14]
```diff
@@ -1,222 +1,223 @@
...
-    </footer>
-
-    <div class="btm-spacer"></div>
+      </div>
+    </div>
+  </div><!-- modal-content [summary] -->
+  <footer ng-if="currentForm === 'summaryForm'">
+    <a ng-click="submit()"
+       data-modal-tab="next" class="submit"
+                             ng-class="{active: bulkDeleteForm.summaryForm.$valid, inactive: bulkDeleteForm.summaryForm.$invalid, disabled: submitButton}">
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/delete-events-modal.html:214]: Weird alignment

**Comment by staubesv** [comment]
Hi @lkiesow 

Basically looks good to me.

The most annoying formatting I've found is things like:

```
<element attribute1="value1" attribute2="value2" attribute3="value3"
                                                 attribute4="For whatever reason not aligned to attribute 1"/>
```

Comments are sometimes aligned like HTML elements and sometimes not, but that is not big deal.

Anyway, that kind of things could be "improved" later on the go...

**Comment by lkiesow** [comment]
@staubesv, the HTML linting is not done at all in the admin-ui module at the moment. There are a lot of HTML syntax errors for which I do not have the time to fix them right now. However, to prevent a huge second pull request, later on, I threw all of them through some auto-formatting with the basic rules.

This pull request also likely needs some more work tomorrow since merging other pull requests will break this again.

**Review by staubesv — APPROVED** [review]
Merge conflicts obviously need to be solved, but then it is ok.

**Pull request merged**